src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -226,7 +241,9 @@ def iot_dps_linked_hub_delete(client, dps_name, resource_group_name, linked_hub)
     dps_property = IotDpsPropertiesDescription(None, None, updated_hub, dps.properties.allocation_policy, dps.properties.authorization_policies)
     dps_description = ProvisioningServiceDescription(dps.location, dps_property, dps.sku)
 
-    client.iot_dps_resource.create_or_update(resource_group_name, dps_name, dps_description)
+    if no_wait:
+        return client.iot_dps_resource.create_or_update(resource_group_name, dps_name, dps_description)
+    LongRunningOperation(cmd.cli_ctx)(client.iot_dps_resource.create_or_update(resource_group_name, dps_name, dps_description))
     return iot_dps_linked_hub_list(client, dps_name, resource_group_name)
```

Comment:
FYI, delete commands in the CLI typically return nothing. However, is this is the behavior you want, that's totally fine.

Reply:
I am thinking still return the list so the user could verify the right entry indeed got deleted. So I think I am keeping it :)